src/sentry/mediators/sentry_apps/updater.py

```diff
@@ -68,6 +70,14 @@ def _update_scopes(self):
         ):
             raise APIError("Cannot update permissions on a published integration.")
         self.sentry_app.scope_list = self.scopes
+        # update the scopes of active tokens tokens
+        tokens = ApiToken.objects.filter(
+            Q(expires_at__isnull=True) | Q(expires_at__gt=timezone.now()),
+            application=self.sentry_app.application,
+        )
+        for token in tokens:
+            token.scope_list = self.scopes
+            token.save()
```

Comment:
You can probably just do `tokens.update(scope_list=self.scopes)` to update them all at once, in one query.

Reply:
yeah I think you can even tack it onto the filter

Reply:
@MeredithAnya @mnoble updated